db/migrate/20171001035221_move_setting_to_tags.rb

```diff
@@ -0,0 +1,23 @@
+class MoveSettingToTags < ActiveRecord::Migration[5.0]
+  def up
+    print "\n"
+    Character.where.not(setting: nil).each do |character|
+      setting = Setting.where(name: character.setting).first
+      setting ||= Setting.create!(user: character.user, name: character.setting)
+      character.settings << setting
```

Comment:
Also I'm wary of adding associations without saving – there are _some_ circumstances (which I think do not apply here) where it doesn't save the change. (`.update_attributes(settings: [setting])` should work the same since the characters only have one setting?)